packages/jobs/lib/execution/action.ts

```diff
@@ -129,10 +132,34 @@ export async function startAction(task: TaskAction): Promise<Result<void>> {
     }
 }
 export async function handleActionSuccess({ taskId, nangoProps, output }: { taskId: string; nangoProps: NangoProps; output: JsonValue }): Promise<void> {
-    const task = await setTaskSuccess({ taskId, output });
-
     const logCtx = logContextGetter.get({ id: nangoProps.activityLogId, accountId: nangoProps.team.id });
+    const { environment, account } = (await environmentService.getAccountAndEnvironment({ environmentId: nangoProps.environmentId })) || {
+        environment: undefined,
+        account: undefined
+    };
 
+    const task = await setTaskSuccess({ taskId, output });
+    if (task.isErr()) {
+        onFailure({
+            connection: {
+                id: nangoProps.nangoConnectionId,
+                connection_id: nangoProps.connectionId,
+                environment_id: nangoProps.environmentId,
+                provider_config_key: nangoProps.providerConfigKey
+            },
+            syncName: nangoProps.syncConfig.sync_name,
+            provider: nangoProps.provider,
+            providerConfigKey: nangoProps.providerConfigKey,
+            activityLogId: nangoProps.activityLogId,
+            runTime: (new Date().getTime() - nangoProps.startedAt.getTime()) / 1000,
+            error: new NangoError('action_script_failure', { error: task.error }),
+            team: account,
+            environment: environment,
+            syncConfig: nangoProps.syncConfig,
+            endUser: nangoProps.endUser
+        });
+        return;
+    }
```

Comment:
not related to webhooks but we were not failing if setting the state didn't work